### ISSUE TITLE
Change all `tf.nn.dropout` to `tf.layers.dropout`

### ISSUE THREAD

**Issue by dpressel** [dpressel/baseline#200]
This will eliminate a lot of the nastiness with having a `pdrop_value` AND a `pkeep` property on each class, and having to remember to do 1 - pdrop_value everwhere.  It will also make TF more consistent with the other frameworks.  Finally, it makes it possible to have the `TRAIN_FLAG()` be a boolean (or boolean placeholder) which is simpler and cleaner.

**Comment by dpressel** [comment]
Tracking this here: https://github.com/dpressel/baseline/compare/feature/v1...feature/tf-layers-dropout?expand=1

**Comment by dpressel** [comment]
PR: https://github.com/dpressel/baseline/pull/204

**Comment by dpressel** [comment]
merged
